tests/integration/security/rbac/rbac_v2_test.go

```diff
@@ -76,39 +84,39 @@ func TestRBACV2(t *testing.T) {
-		{To: appA, From: appC, Port: 90, Protocol: apps.AppProtocolTCP, ExpectedSuccess: false},
-		{To: appA, From: appD, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/", ExpectedSuccess: false},
-		{To: appA, From: appD, Port: 90, Protocol: apps.AppProtocolTCP, ExpectedSuccess: false},
-
-		{To: appB, From: appA, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/xyz", ExpectedSuccess: successIfAuthEnabled},
-		{To: appB, From: appA, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/secret", ExpectedSuccess: false},
-		{To: appB, From: appA, Port: 90, Protocol: apps.AppProtocolTCP, ExpectedSuccess: successIfAuthEnabled},
-		{To: appB, From: appC, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/", ExpectedSuccess: successIfAuthEnabled},
-		{To: appB, From: appC, Port: 90, Protocol: apps.AppProtocolTCP, ExpectedSuccess: successIfAuthEnabled},
-		{To: appB, From: appD, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/", ExpectedSuccess: successIfAuthEnabled},
-		{To: appB, From: appD, Port: 90, Protocol: apps.AppProtocolTCP, ExpectedSuccess: successIfAuthEnabled},
-
-		{To: appC, From: appA, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/", ExpectedSuccess: false},
-		{To: appC, From: appA, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/secrets/admin", ExpectedSuccess: false},
-		{To: appC, From: appA, Port: 90, Protocol: apps.AppProtocolTCP, ExpectedSuccess: false},
-		{To: appC, From: appB, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/", ExpectedSuccess: false},
-		{To: appC, From: appB, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/credentials/admin", ExpectedSuccess: false},
-		{To: appC, From: appB, Port: 90, Protocol: apps.AppProtocolTCP, ExpectedSuccess: false},
-		{To: appC, From: appD, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/", ExpectedSuccess: successIfAuthEnabled},
-		{To: appC, From: appD, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/any_path/admin", ExpectedSuccess: false},
-		{To: appC, From: appD, Port: 90, Protocol: apps.AppProtocolTCP, ExpectedSuccess: false},
-
-		{To: appD, From: appA, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/xyz", ExpectedSuccess: true},
-		{To: appD, From: appA, Port: 90, Protocol: apps.AppProtocolTCP, ExpectedSuccess: false},
-		{To: appD, From: appB, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/", ExpectedSuccess: true},
-		{To: appD, From: appB, Port: 90, Protocol: apps.AppProtocolTCP, ExpectedSuccess: false},
-		{To: appD, From: appC, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/any_path", ExpectedSuccess: true},
-		{To: appD, From: appC, Port: 90, Protocol: apps.AppProtocolTCP, ExpectedSuccess: false},
+		// Port 80 is where HTTP is served, 90 is where TCP is served. When an HTTP request is at port
+		// 90, this means it is a TCP request. The test framework uses HTTP to mimic TCP calls in this case.
+		{To: appA, From: appB, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/xyz", IsAllowed: false},
```

Comment:
nit: a utility function like createConnection(appA, appB, 80, "/xyz", false) is better, at least you don't need to repeat all these `apps.AppProtocolHTTP`.

Reply:
I keep using fields like To, From, Port, etc. mainly because it's easier to reason what it's doing. `createConnection("a", "b", "http", "/xyz")` doesn't tell which one is the src or dst, whether "/xyz" is the path for some token or header value, etc. Are you saying we're passing in 4 strings, rather than a struct (with individual fields) of strings? Sorry if I misinterpreted your comment.

Reply:
This can be solved simply by creating your own test case. 
for example, you can have
```
	type request struct{
		from string
		to string
		path string
		protocol string
	}
	cases := []struct{
		req request
		expectAllow bool
	}{
		{req: request{from: "a", to: "b", path: "/abc", protocol: "http"}, expectAllow: true},
		{req: request{from: "a", to: "c", protocol: "tcp"}, expectAllow: false},
	}
```

It's more readable (you know if it's testing http or tcp) and shorter as you can avoid repeating the `apps.AppProtocolHTTP` and Port.

Reply:
This doesn't address the second reason I talked above, alignment with other security tests (cc @lei-tang). 
It also requires parsing app names, e.g. "a", or getting app multiple times, one per request. 
In the future we may need to use more than just port 80 or 90.

I'm fine with either way. I just don't think create more structs + all the reasons above > reusing the connection struct, unless I'm missing something.

Reply:
I encourage reuse of the data structures in test framework to: 1. avoid duplicate code. 2. reusing makes the test code shorter and less error prone (new code tend to be less battle tested as existing code).